src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -42,22 +42,24 @@ contract MinimalProtectedProxyFactory is SphereXConfiguration {
      *  If the current engine is not the null address, also updates the engine about a valid new sender
      * @dev notice the deploy function itself is NOT PROTECTED
      */
-    function deploy() virtual public onlyAllowedDeployer returns (address proxyAddress) {
-        address engineAddress = sphereXEngine();
+    function deploy() public virtual onlyAllowedDeployer returns (address proxyAddress) {
         SpherexProtetedMinimalProxy minimalProxy = new SpherexProtetedMinimalProxy(
-            sphereXAdmin(), 
-            address(this), // for configuring the allowed sigs 
-            engineAddress, 
+            address(this),
+            address(this), // for configuring the allowed sigs
+            sphereXEngine(),
             IMPLEMENTATION
         );
         proxyAddress = address(minimalProxy);
 
         _addAllowedSenderOnChain(address(proxyAddress));
         minimalProxy.addProtectedFuncSigs(PROTECTED_SIGS);
         // Now that the sigs are set, we can set the real operator
-        minimalProxy.changeSphereXOperator(this.sphereXOperator());
+        minimalProxy.changeSphereXOperator(sphereXOperator());
+
+        // the admin will stay this contract until sphereXAdmin() will
```

Comment:
Is this what normally happen in factories? This might be dangerous in case the factory contains arbitrary call

Reply:
what factory? this is our factory...we dont let anyone inherit from it

Reply:
I meant in the general case where we embed our setup code in the customer factory

Reply:
no of course not, but in the general case the factory doesnt configure the proxy protected sigs like we do here